test/service.go

```diff
@@ -113,3 +116,156 @@ func WaitForOperatorDepsDeleted(ctx *Context) error {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := ctx.Clients.Kube.AppsV1().Deployments(namespace).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx.AddToCleanup(func() error {
+		return DeleteDeployment(ctx, deployment.Name, namespace)
+	})
+
+	kubeService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: 80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+				},
+			},
+			Selector: map[string]string{
+				"app": name,
+			},
+		},
+	}
+
+	svc, err := ctx.Clients.Kube.CoreV1().Services(namespace).Create(kubeService)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx.AddToCleanup(func() error {
+		return DeleteService(ctx, svc.Name, namespace)
+	})
+
+	return svc, nil
+}
+
+func WithRouteForServiceReady(ctx *Context, serviceName, namespace string) (*routev1.Route, error) {
+	r := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: namespace,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: serviceName,
+			},
+		},
+	}
+
+	route, err := ctx.Clients.Route.Routes(namespace).Create(r)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx.AddToCleanup(func() error {
+		return DeleteRoute(ctx, route.Name, namespace)
+	})
+
+	route, err = WaitForRouteState(ctx, route.Name, route.Namespace, RouteHasHost)
+	if err != nil {
+		return nil, err
+	}
+
+	return route, nil
+}
+
+func WaitForRouteState(ctx *Context, name, namespace string, inState func(s *routev1.Route, err error) (bool, error)) (*routev1.Route, error) {
+	var lastState *routev1.Route
+	var err error
+	waitErr := wait.PollImmediate(Interval, Timeout, func() (bool, error) {
+		lastState, err = GetRoute(ctx, name, namespace)
+		return inState(lastState, err)
+	})
+
+	if waitErr != nil {
+		return lastState, errors.Wrapf(waitErr, "OpenShift Route %s is not in desired state, got: %+v", name, lastState)
+	}
+	return lastState, nil
+}
+
+func GetRoute(ctx *Context, name, namespace string) (*routev1.Route, error) {
+	return ctx.Clients.Route.Routes(namespace).Get(name, metav1.GetOptions{})
+}
+
+func RouteHasHost(r *routev1.Route, err error) (bool, error) {
```

Comment:
You might want to actually look for the `RouteAdmitted` condition instead. At least that's what the networking-edge team told me.